sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -54,6 +54,7 @@ import org.apache.spark.sql.util.ExecutionListenerManager
  * @groupname Ungrouped Support functions for language integrated queries
  * @since 1.0.0
  */
+@deprecated("Use SparkSession instead", "3.0.0")
```

Comment:
Adding a deprecation notice per @rxin

Reply:
One more very late question for 2.4 -- @rxin is it even better to deprecate this right now in 2.4.0? and in Python, R?

Reply:
Yea I wouldn't deprecate it now ... the data source API v1 still depends on it. 

Actually now I think about it, we should not be deprecate SQLContext until dsv2 is stable. Otherwise we have a stable API dsv1 depending on a deprecated API.